sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2828,4 +2839,16 @@ class Dataset[T] private[sql](
       Dataset(sparkSession, logicalPlan)
     }
   }
+
+  /** Convert to an RDD of ArrowPayload byte arrays */
+  private[sql] def toArrowPayloadBytes(): RDD[Array[Byte]] = {
+    val schema_captured = this.schema
+    queryExecution.toRdd.mapPartitionsInternal { iter =>
+      val converter = new ArrowConverters
+      val payload = converter.interalRowIterToPayload(iter, schema_captured)
+      val payloadBytes = ArrowConverters.payloadToByteArray(payload, schema_captured)
```

Comment:
This works now by consuming all rows from the iterator at once and constructing a `ArrowPayload` for them. It might harm for memory usage if the rows are huge.

I think a better way might be to only construct a `ArrowPayload` for a group of rows, not all rows.